src/app/frontend/deploy/deploy_settings_controller.js

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {stateName as replicasetliststate} from 'replicasetlist/replicasetlist_state';
+
+/**
+ * Controller for the deploy using settings directive.
+ *
+ * @final
+ */
+export default class DeploySettingsController {
+  /**
+   * @param {!angular.Scope} $scope
+   * @param {!angular.$log} $log
+   * @param {!ui.router.$state} $state
+   * @param {!angular.$resource} $resource
+   * @ngInject
+   */
+  constructor($scope, $log, $state, $resource) {
```

Comment:
I tried using <deploy-from-settings deploy="ctrl.deployFn" /> but when deploy is with deploy button in parent scope triggered, this function can't access to the attributes of the DeploySettingsController class.